### PR TITLE
Add command to add student into existing team in tutorial class.

### LINKED ISSUE #103: Allocate students in teams
As a TA, I can allocate students into teams for class projects and assignments.

Command Format: `/allocate_team id/<student_id> module/<moduleCode> tutorial/<tutorial_class> name/<team_name>`

### PR DESCRIPTION
**Pull request by Zack-Tay**
Fixes #103

Only support adding by id for now. This PR will be further refined to include testcases and email and index addition support.
Add a new command that allows user to add an existing student that is
in the tutorial class to an existing team in that tutorial group.

* Add a new AllocateStudentToTeamCommand and
AllocateStudentToTeamCommandParser class to add the user into an
existing team of an existing tutorial group in an existing module.

* Add more functions to AddressBook class to support the command and
parser above.

### CODE REVIEW

**Review by mahadhir247 — COMMENTED** [review]
Generally LGTM! Just a comment regarding a method in AddressBook.java. Good job abstracting out code to adhere to coding principals.

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -228,6 +245,76 @@ public void deletePersonFromTutorialClass(Person person, ModuleCode module, Tuto
...
+     * Returns true if the {@code studentId} is already in a team of {@code tutorialClass}.
+     * @param tutorialClass of the teams.
+     * @param studentId to search for.
+     */
+    public boolean isStudentInAnyTeam(StudentId studentId, TutorialClass tutorialClass) {
+        boolean isStudentExist = false;
+        for (TutorialTeam tutorialTeam : tutorialClass.getTeams()) {
+            isStudentExist = tutorialTeam.hasStudentVerified(studentId, tutorialTeam);
+        }
+        return isStudentExist;
```
[the remark this answers is not in the document]
Comment (Zack-Tay) [reply src/main/java/seedu/address/model/AddressBook.java:281]: Yes indeed, you are right! I missed that out, thanks for highlighting.

File: src/main/java/seedu/address/logic/commands/allocatestudenttoteamcommands/AllocateStudentToTeamByIndexCommand.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import seedu.address.model.module.TutorialClass;
+import seedu.address.model.module.TutorialTeam;
+import seedu.address.model.person.Person;
+
+/**
+ * Allocates a student to a team in a tutorial Class in TAHelper.
+ */
+public class AllocateStudentToTeamByIndexCommand extends AllocateStudentToTeamCommand {
+
+    public static final String COMMAND_WORD = "/allocate_team";
```
Comment (Jajared) [inline src/main/java/seedu/address/logic/commands/allocatestudenttoteamcommands/AllocateStudentToTeamByIndexCommand.java:25]: I think there are certain variables which can be removed as they have already been defined in the parent abstract class

File: src/main/java/seedu/address/logic/commands/allocatestudenttoteamcommands/AllocateStudentToTeamCommand.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    /**
+     * Check the condition needed to allocate the student to a tutorial team.
+     * @param model
+     * @param student
+     * @param tutClass
+     * @param tutorialTeam
+     * @throws CommandException
+     */
+    public void checkAllocateCondition(Model model, Person student, TutorialClass tutClass, TutorialTeam tutorialTeam)
```
Comment (Jajared) [inline src/main/java/seedu/address/logic/commands/allocatestudenttoteamcommands/AllocateStudentToTeamCommand.java:61]: Good abstraction of this method out to the parent class

File: src/main/java/seedu/address/logic/parser/AllocateStudentToTeamCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_STUDENTID, PREFIX_EMAIL, PREFIX_INDEX,
+                PREFIX_MODULECODE, PREFIX_TUTORIALCLASS, PREFIX_TEAMNAME);
+
+        ModuleCode moduleCode = ParserUtil.parseModuleCode(argMultimap.getValue(PREFIX_MODULECODE).get());
+        TutorialClass tutorialClass = ParserUtil.parseTutorialClass(argMultimap.getValue(PREFIX_TUTORIALCLASS).get());
+        TutorialTeam tutorialTeam = ParserUtil.parseTutorialTeam(argMultimap.getValue(PREFIX_TEAMNAME).get());
+
+        if (doesStudentIdExist) {
```
Comment (Jajared) [inline src/main/java/seedu/address/logic/parser/AllocateStudentToTeamCommandParser.java:59]: This implementation is similar to what I previously implemented for addStudentToClass. It works but just be reminded to state in any documentation on the target priority (i.e what if both id and email are entered, which shall take priority, if any?)

**Review by Jajared — APPROVED** [review]
LGTM! This was a tough PR so good job on completing it!

File: src/main/java/seedu/address/logic/parser/AllocateStudentToTeamCommandParser.java (hunk 4 above)
Comment (Zack-Tay) [reply src/main/java/seedu/address/logic/parser/AllocateStudentToTeamCommandParser.java:59]: Yeap so in this case I believe studentId and email should take precedence over the index.

**Pull request merged**